docs/csharp/csharp-6.md

```diff
@@ -1,17 +1,570 @@
+* [index initializers](#index-initializers)
+* [Extension methods for collection initializers](#extension-add-methods-in-collection-initializers)
+* [Improved overload resolution](#improved-overload-resolution)
+
+The overall effect of these features is that you write more concise code
+that is also more readable. The syntax contains less ceremony for many
+common practices. It's easier to see the design intent with less
+ceremony. Learn these features well, and you'll be more productive,
+write more readable code, and concentrate more on your core features
+than on the constructs of the language.
+
+The remainder of this topic provides details on each these features.
+
+## Auto Property enhancements 
+
+The syntax for auto-properties made it very easy to create properties
+that had simple get and set accessors:
+
+[!code-csharp[ClassicAutoProperty](../../samples/snippets/csharp/new-in-6/oldcode.cs#ClassicAutoProperty)]
+
+However, this simple syntax limited the kinds of designs you could support using
+auto properties. C# 6 improves the auto properties capabilities so that you can use
+them in more scenarios. You'll fall back on the more verbose syntax of declaring
+the backing field by hand, and manipulating that backing field less often.
+
+The new syntax addresses scenarios for read only properties, and for initializing
+the variable storage behind an auto-property.
+
+### Readonly auto-properties
+
+***Readonly auto-properties*** provide a more concise syntax to create
+immutable types. The closest you could get to immutable types
+in earlier versions of C# was to declare private setters:
+
+[!code-csharp[ClassicReadOnlyAutoProperty](../../samples/snippets/csharp/new-in-6/oldcode.cs#ClassicReadOnlyAutoProperty)]
+ 
+Using this syntax, the compiler doesn't ensure that the type really is immutable. It only
+enforces that the `FirstName` and `LastName` properties are not modified from any
+code outisde the class.
+
+Readonly auto-properties enable true readonly behavior. You declare the auto-property
+with only a get accessor:
+
+[!code-csharp[ReadOnlyAutoProperty](../../samples/snippets/csharp/new-in-6/newcode.cs#ReadOnlyAutoProperty)]
+
+The `FirstName` and `LastName` properties can be set only in the body of a constructor:
+
+[!code-csharp[ReadOnlyAutoPropertyConstructor](../../samples/snippets/csharp/new-in-6/newcode.cs#ReadOnlyAutoPropertyConstructor)]
+
+Trying to set `LastName` in another method generates a `CS0200` compilation error:
+
+```csharp
+public class Student
+{
+    public string LastName { get;  }
+
+    public void ChangeName(string newLastName)
+    {
+        // Generates CS 0200: Property or indexer cannot be assigned to -- it is read only
+        LastName = newLastName;
+    }
+}
+```
+
+This features enables true language support for creating immutable types and using
+the more concise and convenient auto-property syntax.
+
+### Auto Property Initializers
```

Comment:
I checked the Roslyn site, and the C# spec. The full name for the feature is "automatically implemented properties". The shortened form is typically hyphenated.  

I made updates to fix both.